### ISSUE TITLE
Error in Modelica.StateGraph.StateGraphRoot annotation

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#353]
**Reported by stefanv on 13 Apr 2010 15:19 UTC**
This model contains the annotation:

```
          Rectangle(
            extent={{-100,100},{100,-100}},
            fillColor=DynamicSelect({255,255,255}, if active > 0.5 then {0,255,
                0} else {255,255,255}),
            lineColor={0,0,0},
            fillPattern=FillPattern.Solid),
```

There is no variable named "active" anywhere in the model or in any base class. Either "active" is an undocumented feature of annotations, scoping rules for variables appearing in annotations are different, or this annotation is incorrect.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/353

**Comment by modelica-trac-importer** [comment]
**Modified by stefanv on 13 Apr 2010 15:36 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by stefanv on 13 Apr 2010 16:09 UTC**
Another possibility is that the variable "active" is missing from StateGraphRoot.

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 22 Mar 2011 10:07 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 7 Mar 2012 15:12 UTC**
Stefan, to me it looks like a simple leftover from the other step components which actually have `active` defined. As I see it the `StateGraphRoot` component would never appear as `active`. So we can just remove that dynamic dependency in the icon layer.

Fixed in affdf44ec69d14a20d87fd47c01156060f9252f3.
